src/plugins/data/server/search/strategies/sql_search/request_utils.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { SqlGetAsyncRequest, SqlQueryRequest } from '@elastic/elasticsearch/lib/api/types';
+import { ISearchOptions } from '../../../../common';
+import { SearchSessionsConfigSchema } from '../../../../config';
+
+/**
+ @internal
+ */
+export function getDefaultAsyncSubmitParams(
+  searchSessionsConfig: SearchSessionsConfigSchema | null,
+  options: ISearchOptions
+): Pick<SqlQueryRequest, 'keep_alive' | 'wait_for_completion_timeout' | 'keep_on_completion'> {
+  const useSearchSessions = searchSessionsConfig?.enabled && !!options.sessionId;
+
+  const keepAlive = useSearchSessions
+    ? `${searchSessionsConfig!.defaultExpiration.asMilliseconds()}ms`
+    : '1m';
+
+  return {
+    // Wait up to 100ms for the response to return
+    wait_for_completion_timeout: '100ms',
+    // If search sessions are used, store and get an async ID even for short running requests.
+    keep_on_completion: useSearchSessions,
+    // The initial keepalive is as defined in defaultExpiration if search sessions are used or 1m otherwise.
+    keep_alive: keepAlive,
+  };
+}
+
+/**
+ @internal
+ */
+export function getDefaultAsyncGetParams(
+  searchSessionsConfig: SearchSessionsConfigSchema | null,
+  options: ISearchOptions
+): Pick<SqlGetAsyncRequest, 'keep_alive' | 'wait_for_completion_timeout'> {
+  const useSearchSessions = searchSessionsConfig?.enabled && !!options.sessionId;
+
+  return {
+    // Wait up to 100ms for the response to return
+    wait_for_completion_timeout: '100ms',
+    ...(useSearchSessions
+      ? // Don't change the expiration of search requests that are tracked in a search session
+        undefined
+      : {
+          // We still need to do polling for searches not within the context of a search session or when search session disabled
+          keep_alive: '1m',
+        }),
+  };
+}
```

Comment:
Is there enough overlap between this and the `ese` request utils file to warrant combining them? It seems like the parameters are almost identical. Do the different request types make this more difficult than it is worth?

Reply:
They are quite different. I think combining them would bring more complexity than benefit

Reply:
Let's do it 🚀 

I suggest we do it separately and review it carefully. Started a draft here https://github.com/elastic/kibana/pull/128358 
I suggest we should extract polling/async/session-related stuff into `common/async_utils` and then use it in other strategies. 

I suggest we merge this and then I finish with the draft

Reply:
🙌   Sounds great